lightning/src/ln/channel.rs

```diff
@@ -4789,7 +4796,12 @@ impl<'a, Signer: Sign, K: Deref> ReadableArgs<&'a K> for Channel<Signer>
 		}
 		let holder_signer = keys_source.read_chan_signer(&keys_data)?;
 
-		let shutdown_pubkey = Readable::read(reader)?;
+		// Read the old serialization for shutdown_pubkey, preferring it for shutdown_scriptpubkey
+		// over the TLV if valid.
+		let mut shutdown_scriptpubkey = match <PublicKey as Readable>::read(reader) {
+			Ok(pubkey) => Some(ShutdownScript::new_p2wpkh_from_pubkey(pubkey)),
```

Comment:
Is it possible for this to read the dummy `[0u8; PUBLIC_KEY_SIZE]` that was written and falsely succeed?

Reply:
IIUC, reading an all-zero pubkey will fail because the prefix byte (indicating whether or not it is in compressed format) would be invalid.